packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -1364,9 +1368,14 @@ function CheckoutLineItem( {
 									.map( ( cart_product ) => cart_product.uuid );
 							}
 
-							Promise.all( product_uuids_to_remove.map( removeProductFromCart ) ).catch( () => {
-								// Nothing needs to be done here. CartMessages will display the error to the user.
-							} );
+							await Promise.all( product_uuids_to_remove.map( removeProductFromCart ) ).catch(
+								() => {
+									// Nothing needs to be done here. CartMessages will display the error to the user.
+								}
+							);
+
+							setRestorableProducts?.( [ ...( restorableProducts || [] ), product ] );
+
 							onRemoveProduct?.( label );
```

Comment:
It's a shame that `onRemoveProduct` only passes the label. We could have used it.

Reply:
Now that we know this is in a package, and we should aim to only minimally alter this package, maybe we can add a second parameter to `onRemoveProduct` and pass the `product`? Then handle the event on the app side? WDYT?

Reply:
Given that the package contains functions and components for use by WordPress.com checkout I think it's fine to have the changes on it.